mongo_orchestration/replica_sets.py

```diff
@@ -59,34 +59,61 @@ def __init__(self, rs_params):
         if not not self.sslParams:
             self.kwargs['ssl'] = True
 
-        config = {"_id": self.repl_id, "members": [
-                  self.member_create(member, index) for index, member in enumerate(rs_params.get('members', {}))
-                  ]}
+        # Always use a keyFile if auth is on.
+        if self.login and not self.auth_key:
+            self.auth_key = DEFAULT_AUTH_KEY
+
+        # Find a server that is suitable to initiate the set.
+        other_members = []
+        members = rs_params.get('members', [])
+        for index, member in enumerate(rs_params.get('members', [])):
+            if not (member.get('arbiterOnly', False)
+                    or member.get('priority', 1) == 0):
+                first_member = index
+                break
+        else:
+            raise ReplicaSetError(
+                'No member suitable to initialize set in %r' % members)
+
+        # Start first server. This will automatically add a user and keyFile.
+        params = members[first_member].get('procParams', {})
+        first_server_id = Servers().create(
+            name='mongod',
+            procParams=members[first_member].get('procParams', {}),
+            sslParams=self.sslParams,
+            auth_key=self.auth_key,
+            login=self.login,
+            password=self.password,
+            version=self._version
+        )
+
+        # Restart first server as a replica set member.
+        def add_repl_set(config):
```

Comment:
Because this may add a user, but you can't add a user to an uninitialized replica set member, since it's "not master." We could start up the whole replica set without auth first, add the user, then restart every machine with auth enabled, but that involves restarting more mongods. This is the least bad option I could come up with.